docs/resources/server/enchantments.md

```diff
@@ -0,0 +1,85 @@
+
+The JSON format for Enchantments is summarized on the [Enchantment definition Minecraft wiki page].
+
+## Enchantment Components
+Enchantment Components are specially-registered [Data Components] that determine what an enchantment does. Enchantment Component Types must be [registered] to `BuiltInRegistries.ENCHANTMENT_EFFECT_COMPONENT_TYPE`.
+:::note
+There are many already existing Enchantment Component Types in vanilla that can be used to create enchantment logic.
+:::
+
+For simple enchantments that act as a marker for custom logic, `DataComponentType<Unit>` may be sufficient. Such a component type could be registered, for example, using a DeferredRegister:
+```java
+// In some registration class
+public static final DeferredRegister<DataComponentType<?>> ENCHANTMENT_COMPONENT_TYPES = DeferredRegister.create(BuiltInRegistries.ENCHANTMENT_EFFECT_COMPONENT_TYPE, "examplemod");
+
+public static final DeferredHolder<DataComponentType<?>, DataComponentType<Unit>> EXAMPLE =
+    ENCHANTMENT_COMPONENT_TYPES.register("example",
+        () -> DataComponentType.<Unit>builder()
+            .persistent(Unit.CODEC)
+            .build());
+```
+
+### Testing For Enchantment Components
+To check if an enchantment component is present on a given item stack, one must first get the item's `minecraft:enchantment` component. The value of this component is an `ItemEnchantments` instance containing a set of all the Enchantments on the item, which can be acquired using `ItemEnchantments#keySet()`. The enchantment components of a given enchantment can then be acquired using the `Enchantment::effects()` method:
+```java
+
+for(Enchantment enchant : stack.get(DataComponents.ENCHANTMENTS).keySet()){
+    DataComponentMap enchantment_components = enchant.value().effects();
+    if(enchantment_components.has(MY_ENCHANTMENT_COMPONENT_TYPE_HOLDER)){
+            // Do something!
+    }
+}
+```
+
+## Specialized Enchantment Component Types
+Vanilla uses a few specific classes when building its enchantment component types to achieve the customizability present in the vanilla enchantment components.
+
+### `ConditionalEffect`
+Wrapping the type in `List<ConditionalEffect<?>>` permits the use of [predicates] when defining enchantments. This allows some effects to activate situationally. Most vanilla enchantments use this.
+Registering such an effect can be done as follows:
+```java
+public static final DeferredHolder<DataComponentType<?>, DataComponentType<List<ConditionalEffect<Unit>>>> EXAMPLE_CONDITIONAL_EFFECT =
+    ENCHANTMENT_COMPONENT_TYPES.register("example_conditional",
+        () -> DataComponentType.<List<ConditionalEffect<Unit>>>builder()
```

Comment:
This should not be a unit with a list as there would be no point to the list. There's only one state the conditional can be in: true or false.